sklearn/preprocessing/_encoders.py

```diff
@@ -414,8 +721,10 @@ def fit(self, X, y=None):
         self
         """
         self._validate_keywords()
-        self._fit(X, handle_unknown=self.handle_unknown,
-                  force_all_finite='allow-nan')
+        fit_results = self._fit(X, handle_unknown=self.handle_unknown,
+                                force_all_finite='allow-nan',
+                                return_counts=self._infrequent_enabled)
+        self._fit_infrequent_category_mapping(fit_results)
         self.drop_idx_ = self._compute_drop_idx()
```

Comment:
could pass `fit_results` to `_compute_drop_idx()` such that we can leverage the `category_counts` for the `most_frequent` drop method in #18679

Reply:
We can change the signature of `_compute_drop_idx` if we need to (as in #18679)